### PR TITLE
Create REd Black trees

### PR DESCRIPTION
**Pull request by pb1826**

### CODE REVIEW

File: Java/REd Black trees [hunk 1]
```diff
@@ -0,0 +1,188 @@
+/*
```
Comment (nandukalidindi) [inline Java/REd Black trees]: No spaces in file names please. File extension missing.

Comment (pb1826) [inline Java/REd Black trees]: oh sorry. 
sure. shall i make a new file and pull another request ?

Comment (nandukalidindi) [inline Java/REd Black trees]: You can just rename this file and push a new commit to this PR.

Comment (pb1826) [inline Java/REd Black trees]: yeah done, thanks.
anything further ?

**Comment by xuzebin** [comment]
1. For method names, please use camelCase with the first letter lowercase. 
2. For constant variables like RED, BLACK, please declare them as static final.

**Comment by xuzebin** [comment]
3. Always use braces even when there is only one statement.
4. No line break before the opening brace.
(You can reference more here: https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used)

**Comment by pb1826** [comment]
Sure , thank you.
I will take care the next time.

**Pull request merged**